app/domain/authentication/jwt/decoded_credentials.rb

```diff
@@ -4,23 +4,44 @@ module Jwt
     class DecodedCredentials
 
       JWT_REQUEST_BODY_FIELD_NAME = "jwt".freeze
+      # https://datatracker.ietf.org/doc/html/rfc4648#section-5
+      # jwt token is 3 blocks of base64url encoded strings separated by period '.'
+      # base64url encoding differs from regular base64 encoding as follows
+      # - padding is skipped so the pad character '=' doesn't have to be percent encoded
+      # - the 62nd and 63rd regular base64 encoding characters ('+' and '/') are replace with ('-' and '_')
+      #   The changes make the encoding alphabet file and URL safe.
+      # tokens without a signature part are denied
+      BASE_64_URL_REGEX = "[-A-Za-z0-9_=]+".freeze
+      JWT_REGEX = /\A#{BASE_64_URL_REGEX}\.#{BASE_64_URL_REGEX}\.#{BASE_64_URL_REGEX}\z/.freeze
 
       def initialize(credentials)
+        @logger = Rails.logger
         @decoded_credentials = Hash[URI.decode_www_form(credentials)]
-        validate
+        validate_jwt_request_field_is_present
+        validate_body_contains_jwt
       end
 
       def jwt
-        @decoded_credentials[JWT_REQUEST_BODY_FIELD_NAME]
+        @jwt
       end
 
       private
 
-      def validate
+      def validate_jwt_request_field_is_present
         if @decoded_credentials.fetch(JWT_REQUEST_BODY_FIELD_NAME, "") == ""
           raise Errors::Authentication::RequestBody::MissingRequestParam, JWT_REQUEST_BODY_FIELD_NAME
         end
+        @jwt = @decoded_credentials[JWT_REQUEST_BODY_FIELD_NAME].strip
       end
```

Comment:
the method name is `validate_jwt_request_field_is_present` but you also update the `jwt` param value. looks like you need to separate this.

Reply:
do we really need line 34? what's the problem with the current implementation?

Reply:
The `jwt` parameter value should be trimmed before regex validation. Our automation sends the value with `\n` at the end.
And in general it's better to return the exact value without additional characters

Reply:
ok so add `.strip` to the original `jwt` method

Reply:
`.strip` should be added before regex validation
I want to avoid adding it twice.

Reply:
i don't understand. what if you have the method:
```
def jwt
  @jwt ||= @decoded_credentials[JWT_REQUEST_BODY_FIELD_NAME].strip
end
```

and then call `jwt` in line 42. wouldn't it do the job?

Reply:
will extract the line to separate function